### PR TITLE
Reorganize test_execute.py

### PR DESCRIPTION
**Pull request by joannfeng**

### CODE REVIEW

File: test/test_execute.py [hunk 1]
```diff
@@ -1863,131 +1714,134 @@ def test_output_from(self):
...
-        self.assertEqual(res['__completed__']['__step_skipped__'], 0)
+        assert res['__completed__']['__step_completed__'] == 2
+        assert res['__completed__']['__step_skipped__'] == 0
         #
         os.remove('a.bam')
         res = Base_Executor(wf, config={'trace_existing': True}).run()
-        self.assertEqual(res['__completed__']['__step_completed__'], 2)
-        self.assertEqual(res['__completed__']['__step_skipped__'], 1)
+        assert res['__completed__']['__step_completed__'] == 2
+        assert res['__completed__']['__step_skipped__'] == 1
```
Comment (joannfeng) [inline test/test_execute.py:1811]: I tried to make the script variable a string and change the line 
`res = Base_Executor(wf, config={'trace_existing': True}).run()`
to
`res = execute_workflow(script,config={'trace_existing': True})`

But this resulted in a failed assertion statement. Is my syntax incorrect?

Comment (BoPeng) [reply test/test_execute.py:1811]: I think in `execute_workflow` we use `options` for such things and `config` for replacing user-specified config file.

Comment (BoPeng) [reply test/test_execute.py:1811]: https://github.com/vatlab/sos/blob/master/src/sos/__init__.py#L47

**Comment by joannfeng** [comment]
This is a list of the tests that are currently failing:
```
FAILED test_execute.py::test_dynamic_input - AssertionError: False
FAILED test_execute.py::test_output_report_with_dag - assert 'Execution DAG' ...
FAILED test_execute.py::test_rerun_with_zap - FileNotFoundError: [Errno 2] No...
FAILED test_execute.py::test_reexecution_of_dynamic_depends - assert 1 == 2
FAILED test_execute.py::test_traced_function - assert 1 == 2
FAILED test_execute.py::test_error_handling_of_substeps - AssertionError: ass...
FAILED test_execute.py::test_error_handling_of_tasks - AssertionError: assert...

```

**Comment by BoPeng** [comment]
I have fixed the tests using the attached patch (apply using `git apply`). Note that the `os.remove` statements between the tests removes some files to trigger the execution of the workflow and should not be removed.

[a.txt](https://github.com/vatlab/sos/files/4886117/a.txt)

**Comment by BoPeng** [comment]
@joannfeng I accidentally merged the PR and had to apply the patch. Please synchronize with the master if needed.

**Comment by joannfeng** [comment]
Thank you! The patch fixed most of the errors but these three tests are still failing from my end:
```
FAILED test_execute.py::test_dynamic_input - AssertionError...
FAILED test_execute.py::test_output_report_with_dag - asser...
FAILED test_execute.py::test_rerun_with_zap - FileNotFoundE...
```
Is this fine? I will get started on the other files.

**Comment by BoPeng** [comment]
@joannfeng These tests pass in my environment and in travis/appveyor (https://travis-ci.org/github/vatlab/sos/jobs/705988251), which means there is something missing in your environment.

This is generally ok because these tests test a lot of things and do not have to pass everywhere. However, you should have run them before you change them and known if they are supposed to pass in your environment, and you can skip tests that do not pass.

**Pull request merged**